### ISSUE TITLE
Investigating incorrect indentation by google-java-formatter

### ISSUE THREAD

**Issue by Zopsss** [checkstyle/checkstyle#15336]
I have read check documentation: https://checkstyle.org/checks/misc/indentation.html
I have downloaded the latest checkstyle from: https://checkstyle.org/cmdline.html#Download_and_Run
I have executed the cli and showed it below, as cli describes the problem better than 1,000 words

Detected at: https://github.com/checkstyle/checkstyle/pull/15332#pullrequestreview-2190042898

When formatting the input files of indentation sections ( 4.2, 4.5.2, 4.8.4.1 ) as part of #15218, some new violations were introduced after the inputs were formatted by the formatter. Below I have showed an example of original code and formatted code with outputs.

**Original Code:**
```java
/** Testing. */
public class TestingIndentationOg  //indent:0 exp:0
    implements Runnable, Cloneable {
  /** Testing. */
  public TestingIndentationOg() { //indent:2 exp:2

    String someString = ""; //indent:4 exp:4

    doNothing(//indent:4 exp:4
        someString //indent:8 exp:8
            .concat("zyx" //indent:12 exp:12
            ) //indent:12 exp:12
            .concat("255, 254, 253" //indent:12 exp:12
            ) //indent:12 exp:12
    ); //indent:4 exp:4

  } //indent:2 exp:2

  class InnerClass2  //indent:2 exp:2
       extends  //indent:7 exp:>=6
          TestingIndentationOg {} //indent:10 exp:>=6

  /******* Helper Methods. *******/

  public String doNothing(String something) { //indent:2 exp:2
    return something; //indent:4 exp:4
  } //indent:2 exp:2

  @Override
  public void run() {
    // TODO Auto-generated method stub
    throw new UnsupportedOperationException("Unimplemented method 'run'");
  }  //indent:4 exp:4

}
```

**Output:**
```
$ java -jar .\checkstyle-10.17.0-all.jar -c .\google_checks.xml .\TestingIndentationOg.java
Starting audit...
Audit done.
```

**Formatted Code:**
```java
/** Tetsing. */
public class TestingIndentationFormtted // indent:0 exp:0
implements Runnable, Cloneable { // indent:4 exp:4 // warn
  /** Tetsing. */
  public TestingIndentationFormtted() {

    String someString = "";

    doNothing(// indent:4 exp:4
        someString // indent:8 exp:8
            .concat(
                "zyx" // indent:12 exp:12
                ) // indent:12 exp:12
            .concat(
                "255, 254, 253" // indent:12 exp:12
                ) // indent:12 exp:12
        ); // indent:4 exp:4 // warn
  }

  class InnerClass2 // indent:2 exp:2
  extends // indent:7 exp:>=6 // warn
      TestingIndentationFormtted {} // indent:10 exp:>=6

  /******* Helper Methods. *******/

  public String doNothing(String something) {
    return something;
  }

  @Override
  public void run() {
    // TODO Auto-generated method stub
    throw new UnsupportedOperationException("Unimplemented method 'run'");
  }
}
```

**Output:**
```
$ java -jar .\checkstyle-10.17.0-all.jar -c .\google_checks.xml .\TestingIndentationFormtted.java
Starting audit...
[WARN] C:\checkstyle testing\TestingGoogleStyle\.\TestingIndentationFormtted.java:3:1: 
'implements' has incorrect indentation level 0, expected level should be 4. [Indentation]
[WARN] C:\checkstyle testing\TestingGoogleStyle\.\TestingIndentationFormtted.java:17:9:
 'method call rparen' has incorrect indentation level 8, expected level should be 4. [Indentation]
[WARN] C:\checkstyle testing\TestingGoogleStyle\.\TestingIndentationFormtted.java:21:3: 
'extends' has incorrect indentation level 2, expected level should be 6. [Indentation]
Audit done.
```
---

The `TestingIndentationFormtted.java`'s code is formatted by the google formatter and it is giving violations. I think the formatter is wrong here. The violations are because of section [4.5.2 Indent continuation lines at least +4 spaces](https://google.github.io/styleguide/javaguide.html#s4.5.2-line-wrapping-indent):

> When line-wrapping, each line after the first (each continuation line) is indented at least +4 from the original line.

The rule states that after line-wrapping, each line should be indented at least +4, but formatter didn't increased the indentation for the following lines:

Line No. 3: `implements Runnable, Cloneable { // indent:4 exp:4 // warn`
and
Line No. 21: `extends // indent:7 exp:>=6 // warn`

In the violation message, it is clearly mentioned to increase the indentation by 4 levels. So I guess the violations are correct here.

For line no 17 violation: `); // indent:4 exp:4 // warn`

The closing bracket ( `);` ) is of method call `doNothing`, so the `);` should be indented same level as `doNothing`. This violation message is also correct IMO.

---

**Describe what you expect in detail.**

The violation are caused by rule [4.5.2 Indent continuation lines at least +4 spaces](https://google.github.io/styleguide/javaguide.html#s4.5.2-line-wrapping-indent). I think the formatter is wrong here and it incorrectly indented the code which is why our config is complaining but I would still like know @romani and @rdiachenko's opinion on this before moving forward.

**Comment by romani** [comment]
@Zopsss , please remove from code snippets all comments, no body write code like this and this can confuse formatter.

> TestingIndentationFormtted.java:3:1: 'implements' has

>TestingIndentationFormtted.java:21:3: 'extends' has in

please create or find issue on https://github.com/google/google-java-format/issues , it is line wrap, so indentation should be +4.

>TestingIndentationFormtted.java:17:9: 'method call rparen' 

this is defect on Checkstyle side

**Comment by Zopsss** [comment]
> @Zopsss , please remove from code snippets all comments, no body write code like this and this can confuse formatter.

If I removed the trailing comments, formatter would make the code one-liner like:

```java
  class InnerClass2 extends TestingIndentationFormtted {} 
```

 I guess the formatter did get confused by those comments as you said.

> this is defect on Checkstyle side

I think it isn't defect. The method call was at column 4 so its closing bracket should also be at 4th column.

**Comment by romani** [comment]
this is defect https://github.com/google/google-java-format/issues/1126

but problem is in trailing comment that is rare usage in real code.
Workaround is to remove trailing comments

**Comment by romani** [comment]
I am closing this issue as we do not use `//indent` comments any more for Indentation module in Inputs, we are not affected.
